extensions/Worker.Extensions.ServiceBus/src/ServiceBusMessageActions.cs

```diff
@@ -0,0 +1,168 @@
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Messaging.ServiceBus;
+using Google.Protobuf.WellKnownTypes;
+using Microsoft.Azure.Functions.Worker.Converters;
+using Microsoft.Azure.ServiceBus.Grpc;
+
+namespace Microsoft.Azure.Functions.Worker
+{
+    [InputConverter(typeof(ServiceBusMessageActionsConverter))]
+    public class ServiceBusMessageActions
+    {
+        private readonly Settlement.SettlementClient _settlement;
+
+        internal ServiceBusMessageActions(Settlement.SettlementClient settlement)
+        {
+            _settlement = settlement;
+        }
+
+        ///<inheritdoc cref="ServiceBusReceiver.CompleteMessageAsync(ServiceBusReceivedMessage, CancellationToken)"/>
+        public virtual async Task CompleteMessageAsync(
+            ServiceBusReceivedMessage message,
+            CancellationToken cancellationToken = default)
+        {
+            await _settlement.CompleteAsync(new() { Locktoken = message.LockToken}, cancellationToken: cancellationToken);
+        }
+
+        ///<inheritdoc cref="ServiceBusReceiver.CompleteMessageAsync(ServiceBusReceivedMessage, CancellationToken)"/>
+        public virtual async Task AbandonMessageAsync(
+            ServiceBusReceivedMessage message,
+            IDictionary<string, object>? properties,
+            CancellationToken cancellationToken = default)
+        {
+            var request = new AbandonRequest()
+            {
+                Locktoken = message.LockToken,
+                PropertiesToModify = { TransformProperties(properties) }
+            };
+            await _settlement.AbandonAsync(request, cancellationToken: cancellationToken);
+        }
+
+        ///<inheritdoc cref="ServiceBusReceiver.DeadLetterMessageAsync(ServiceBusReceivedMessage, IDictionary{string, object}, string, string, CancellationToken)"/>
+        public virtual async Task DeadLetterMessageAsync(
+            ServiceBusReceivedMessage message,
+            Dictionary<string, object>? propertiesToModify = default,
+            string? deadLetterReason = default,
+            string? deadLetterErrorDescription = default,
+            CancellationToken cancellationToken = default)
+        {
+            var request = new DeadletterRequest()
+            {
+                Locktoken = message.LockToken,
+                PropertiesToModify = { TransformProperties(propertiesToModify) }
+            };
+            if (deadLetterReason != null)
```

Comment:
is this check necessary? Is there any issue assigning regardless of null or not? Same with line 62..65

Reply:
Yes, the generated code throws if you set to null.

Reply:
Ah left a comment on the proto file - should use `google.protobuf.StringValue` instead.